format/view-spec.md

```diff
@@ -160,6 +179,57 @@ Each entry in `version-log` is a struct with the following fields:
 | _required_  | `timestamp-ms` | Timestamp when the view's `current-version-id` was updated (ms from epoch) |
 | _required_  | `version-id`   | ID that `current-version-id` was set to |
 
+#### Storage Table Identifier
+
+The table identifier for the storage table that stores the precomputed results.
+
+| Requirement | Field name     | Description |
+|-------------|----------------|-------------|
+| _optional_  | `catalog`      | A string specifying the name of the catalog. If set to `null`, the catalog is the same as the view's catalog |
+| _required_  | `namespace`    | A list of strings for namespace levels |
+| _required_  | `name`         | A string specifying the name of the table/view |
```

Comment:
sorry I'm late to this discussion and this might be discussed/could be a stupid question - do we think if it's worth also include the snapshot id of the storage table, so that under circumstances where the storage table was updated but the materialized view was not or vice versa (race condition or commit partial succeeded), there's a reference link in both way (I know storage table already points to view version id under "refresh state"), which might be useful especially when view/table undergo version upgrades that could result in e.g. schema mismatch between the view and the underlying storage table?

Reply:
The current spec is designed such that the materialized view does not need to be updated when the storage table is updated. When the materialized view is refreshed only the storage table is updated. The idea is that the definition of the materialized view is still correct after the refresh.
However, if the materialized view is updated, the storage table should be in the `invalid` state. That's why the `view-version` is stored as part of the `refresh-state`. When the materialized view is updated, it cannot be known if the data in the storage table is still correct. That's why a full refresh has to be performed for an `invalid` state.